src/osversion/VersionTranslator.cpp

```diff
@@ -7664,6 +7665,239 @@ namespace osversion {
+          // * 25 - Supply Air Flow Rate Per Floor Area during Heating Operation
+          // * 26 - Fraction of Autosized Design Heating Supply Air Flow Rate
+          // * 27 - Design Supply Air Flow Rate Per Unit of Capacity During Heating Operation
+          bool found = false;
+          for (size_t i = 0; i < heatingSAFMethodChoices.size(); ++i) {
+            const size_t fieldIndex = heatingSAFMethodIndex + 1 + i;
+            if ((value = object.getString(fieldIndex))) {
+              newObject.setString(heatingSAFMethodIndex, std::string{heatingSAFMethodChoices[i]});
+              newObject.setString(fieldIndex, *value);
+              found = true;
+              break;
+            }
+          }
+          if (!found) {
+            LOG(Warn, "For AirLoopHVACUnitarySystem '"
+                        << object.nameString()
+                        << "', there is a heating coil, but no Supply Air Flow Rate Method During Heating Operation and no flow rate fields set. "
+                           "Defaulting to SupplyAirFlowRate with Autosize");
+            newObject.setString(heatingSAFMethodIndex, "SupplyAirFlowRate");
+            newObject.setString(heatingSAFMethodIndex + 1, "Autosize");
+          }
+        } else {
+          std::string heatingSAFMethodUC = ascii_to_upper_copy(*heatingSAFMethod);
+          if (heatingSAFMethodUC == "NONE") {
+            LOG(Warn, "For AirLoopHVACUnitarySystem '"
+                        << object.nameString()
+                        << "', Supply Air Flow Method Rate During Heating Operation is 'None' but you have a Heating coil, consider changing it");
+          } else {
+            auto it = std::find_if(heatingSAFMethodChoicesUC.cbegin(), heatingSAFMethodChoicesUC.cend(),
+                                   [&heatingSAFMethodUC](auto& s) { return s == heatingSAFMethodUC; });
+            if (it == heatingSAFMethodChoicesUC.cend()) {
+              LOG(Error, "For AirLoopHVACUnitarySystem '"
+                           << object.nameString() << "', Unrecognized Supply Air Flow Method Rate During Heating Operation=" << *heatingSAFMethod);
+            } else {
+              auto dist = std::distance(heatingSAFMethodChoicesUC.cbegin(), it);
+              const size_t index = heatingSAFMethodIndex + 1 + dist;
+              if ((value = object.getString(index))) {
+                newObject.setString(index, *value);
+              } else {
+                LOG(Error, "For AirLoopHVACUnitarySystem '" << object.nameString() << "', Supply Air Flow Method Rate During Heating Operation is '"
+                                                            << *heatingSAFMethod << "' but associated field is empty. Setting it to zero.");
+                newObject.setDouble(index, 0.0);
+              }
+            }
+          }
+        }
+
+        if (!hasHeatingCoil && !hasCoolingCoil) {
+          // TODO: not sure here
+          newObject.setString(noCoolHeatSAFMethodIndex, "None");
```

Comment:
No heating nor cooling coil => set the NoCoolHeatSAFMethod to None. Not sure that's what we want, but it's also what I do at model time.